### PR TITLE
feat(rnd): Add Agent Block sample/test input output for auto-generated unit testing for block

### PR DESCRIPTION
**Pull request by majdyz**
### Background

Sample test input and output on the block can serve as documentation and auto-generated unit-testing code for the agent block.

What's within the scope of this change:
Adding the fields for block test (input, output, mocks), and its execution.

What's still outside the scope:
Handling of mock and stub for a block using sensitive credentials or network calls or 3rd-party connections.

### Changes 🏗️

* Add `test_input` and `test_output` for `Block` class
* Add a unit test that executes all `Block` by using the provided `test_input` and `test_output`.

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/data/block.py [hunk 1]
```diff
@@ -8,7 +8,9 @@
 
 from autogpt_server.util import json
 
-BlockData = dict[str, Any]
+BlockInput = dict[str, Any]
+BlockData = tuple[str, Any]
+BlockOutput = Generator[BlockData, None, None]
```
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/data/block.py:13]: Adding type definitions adds another layer of complexity for understanding what types are being used. 

I'd prefer just using the types directly.

File: rnd/autogpt_server/autogpt_server/server/server.py [hunk 2]
```diff
@@ -147,19 +152,40 @@ def execution_manager_client(self) -> ExecutionManager:
...
+    @classmethod
+    def execute_graph_block(cls, block_id: str, data: dict[str, Any]) -> list:
+        obj = block.get_block(block_id)
+        if not obj:
+            raise HTTPException(status_code=404, detail=f"Block #{block_id} not found.")
+        return [{name: data} for name, data in obj.execute(data)]
 
-    async def get_graphs(self) -> list[str]:
+    @classmethod
+    async def get_graphs(cls) -> list[str]:
```
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/server/server.py:177]: Why are we changing these to class methods?

Comment (majdyz) [reply rnd/autogpt_server/autogpt_server/server/server.py:177]: linter keeps asking to take the function out of the class or make it static.

File: rnd/autogpt_server/autogpt_server/data/block.py (hunk 1 above)
Comment (majdyz) [reply rnd/autogpt_server/autogpt_server/data/block.py:13]: This type can be replaced to be an actual class anytime soon, so keeping it named is really required for this.

image

If you hover on types, you can easily see the type.

**Review by Swiftyos — APPROVED** [review]

**Pull request merged**